sasdata/postprocess.py

```diff
@@ -14,3 +19,49 @@ def apply_fixes(data: SasData, mantid_unit_error=True):
         data = fix_mantid_units_error(data)
 
     return data
+
+
+def deduce_qz(data: SasData):
+    """Calculates and appends Qz to SasData if Qx, Qy, and wavelength are all present"""
+    # if Qz is not already in the dataset, but Qx and Qy are
+    if 'Qz' not in data._data_contents and 'Qx' in data._data_contents and 'Qy' in data._data_contents:
+        # we start by making the approximation that qz=0
+        data._data_contents['Qz'] = 0*data._data_contents['Qx']
+
+        # now check if metadata has wavelength information
+        wavelength = getattr(
+            getattr(
+                getattr(
+                    getattr(data, "metadata", None),
+                    "instrument",
+                    None
+                ),
+                "source",
+                None
+            ),
+            "wavelength",
+            None
+        )
+
+        if wavelength is not None:
+            # we can deduce the value of qz from qx and qy
+            # if we have the wavelength
+            qx = data._data_contents['Qx']
+            qy = data._data_contents['Qy']
+
+            # this is how you convert qx, qy, and wavelength to qz
+            k0 = 2*np.pi/wavelength
+            qz = k0-(k0**2-qx**2-qy**2)**(0.5)
+
+            data._data_contents['Qz'] = qz
+
+if __name__ == "__main__":
+    from sasdata.temp_hdf5_reader import load_data
+    test_file = "/Users/pmneves/Downloads/108854.nxcansas.h5"
```

Comment:
We should not be adding code to the repo containing local directory paths. Is the `if __name__ == "__main__":` section actually necessary here?